pkg/controller/cloudsearch/domain/setup.go

```diff
@@ -0,0 +1,215 @@
+	})
+	if err != nil {
+		return errors.Wrap(err, "failed to retrieve service access policies")
+	}
+
+	spec.AccessPolicies = awsclients.LateInitializeStringPtr(spec.AccessPolicies, respAccessPolicies.AccessPolicies.Options)
+
+	return nil
+}
+
+func (e *external) isUpToDateForScalingParameters(ctx context.Context, cr *svcapitypes.Domain, domainName *string) (bool, error) {
+	resp, err := e.client.DescribeScalingParametersWithContext(ctx, &svcsdk.DescribeScalingParametersInput{
+		DomainName: domainName,
+	})
+	if err != nil {
+		return true, errors.Wrap(err, "failed to retrieve scaling parameters")
+	}
+
+	spec := cr.Spec.ForProvider.CustomDomainParameters
+	stat := resp.ScalingParameters.Options
+
+	// TODO discuss: could be replaced by checking the JSON patch
+	isUpToDate := awsclients.Int64Value(spec.DesiredReplicationCount) == awsclients.Int64Value(stat.DesiredReplicationCount) &&
+		awsclients.StringValue(spec.DesiredInstanceType) == awsclients.StringValue(stat.DesiredInstanceType) &&
+		awsclients.Int64Value(spec.DesiredPartitionCount) == awsclients.Int64Value(stat.DesiredPartitionCount)
+
+	return isUpToDate, nil
+}
+
+func (e *external) isUpToDateAccessPolicies(ctx context.Context, cr *svcapitypes.Domain, domainName *string) (bool, error) {
+	in := svcsdk.DescribeServiceAccessPoliciesInput{
+		DomainName: domainName,
+		Deployed:   awsclients.Bool(false), // include pending policies as well
+	}
+	resp, err := e.client.DescribeServiceAccessPoliciesWithContext(ctx, &in)
+	if err != nil {
+		return true, errors.Wrap(err, "failed to retrieve scaling parameters")
+	}
+
+	spec := cr.Spec.ForProvider.CustomDomainParameters
+	stat := resp.AccessPolicies
+
+	isUpToDate := awsclients.IsPolicyUpToDate(spec.AccessPolicies, stat.Options) && !awsclients.BoolValue(stat.Status.PendingDeletion)
+
+	return isUpToDate, nil
+}
+
+func (e *external) postObserveImpl(ctx context.Context, cr *svcapitypes.Domain, obj *svcsdk.DescribeDomainsOutput, obs managed.ExternalObservation, err error) (managed.ExternalObservation, error) {
+	if err != nil {
+		return managed.ExternalObservation{}, err
+	}
+
+	ds := obj.DomainStatusList[0]
+
+	if awsclients.BoolValue(ds.Deleted) {
+		cr.SetConditions(xpv1.Deleting())
+	} else if awsclients.BoolValue(ds.Created) && ds.SearchService.Endpoint != nil && ds.DocService.Endpoint != nil {
+		if awsclients.BoolValue(ds.Processing) {
+			cr.SetConditions(xpv1.Available().WithMessage("currently processing"))
+		} else {
+			cr.SetConditions(xpv1.Available())
+		}
+	} else {
+		cr.SetConditions(xpv1.Creating())
+	}
+
+	obs.ConnectionDetails = managed.ConnectionDetails{
+		// TODO verify
+		"docServiceEndpoint":    []byte(awsclients.StringValue(ds.DocService.Endpoint)),
+		"searchServiceEndpoint": []byte(awsclients.StringValue(ds.SearchService.Endpoint)),
+	}
+
+	// TODO discuss: correct approach? loop through list of functions instead?
+	scalingParametersUpToDate, err := e.isUpToDateForScalingParameters(ctx, cr, ds.DomainName)
```

Comment:
Uptodate checks should be part of `isUpToDate` and not `postObserve`

Reply:
Yes, I fully agree. The problem I see is that `Context` is not available inside the `isUpToDate` hook. I figure it makes sense to have as many as possible (if not all) sub-calls with the parent context. What is your opinion on that matter? 

I see the following options:
(1) Current approach: Make sub-calls in `postObserve`and modify `ResourceUpToDate` afterwards
(2) Use sub-calls without context in `isUpToDate`
(3) Use sub-calls in `preObserve` and write the results to the extended Domain status model (how would one do that?)
(4) Use sub-calls in `preObserve`and use the hook or external struct to buffer the results (discouraged statefulness)

(3) and (4) would also allow for the `lateInitialize` hook to initialize parameters without contextless & redundant calls.